packages/flutter/lib/src/rendering/layer.dart

```diff
@@ -679,6 +760,23 @@ class ContainerLayer extends Layer {
     assert(transform != null);
   }
 
+  /// Returns the descendants of this layer in depth first order.
+  @visibleForTesting
+  List<Layer> depthFirstIterateChildren() {
+    if (firstChild == null)
+      return null;
+    final List<Layer> children = <Layer>[];
+    Layer child = firstChild;
+    while(child != null) {
+      children.add(child);
+      if (child is ContainerLayer) {
+        children.addAll(child.depthFirstIterateChildren());
```

Comment:
Looks like depthFirstIterateChildren can return null. Will that ever be a problem?

Reply:
Yup! :)